lib/parallel_tests/cucumber/scenarios.rb

```diff
@@ -10,22 +11,25 @@ module Cucumber
     class Scenarios
       class << self
         def all(files, options={})
-          tags = []
-          tags.concat options[:ignore_tag_pattern].to_s.split(/\s*,\s*/).map {|tag| "~#{tag}" }
-          tags.concat options[:test_options].to_s.scan(/(?:-t|--tags) (~?@[\w,~@]+)/).flatten
+          # Combine and generate a tag expression for given test options and ignore tag pattern. Refer here to understand how new tag expression syntax works - https://github.com/cucumber/cucumber/tree/master/tag-expressions
+          tags = options[:test_options].to_s.scan(/(?:-t|--tags) ([^"]*)/).flatten.map { |tag| tag.delete("'")}
```

Comment:
yeah this gets into parsing ... need to be smarter :D 

try
```ruby
require 'shellwords'
tags = []
words = options[:test_options].to_s.shellsplit
words.each_with_index { |w,i| tags << words[i+1] if ["-t", "--tags"].include?(w) }
```

Reply:
Perfect! The previous approach worked too but I agree it looked very complex. `shellsplit` works better in this case. I updated the PR per this suggestion. All rspec tests are passing and I ran some parallel cucumber tests using local gem and it works fine.

I also updated the tag expression in `scenarios_spec.rb` as `cucumber` doesn't support complex tag expressions being passed without single quotes. It only works for single tags now. For example: `cucumber features/  -t @colors` will work but `cucumber features/  -t @colors and not @black` won't work, it has to be `cucumber features/  -t '@colors and not @black'`

I guess this addresses the last issue in PR and we should be good to merge now.